Mage/src/main/java/mage/game/stack/SpellStack.java

```diff
@@ -82,6 +83,7 @@ public boolean counter(UUID objectId, Ability source, Game game, PutCards putCar
             if (!game.replaceEvent(GameEvent.getEvent(GameEvent.EventType.COUNTER, objectId, source, stackObject.getControllerId()))) {
                 if (!(stackObject instanceof Spell)) { // spells are removed from stack by the card movement
                     this.remove(stackObject, game);
+                    game.rememberLKI(Zone.STACK, stackObject);
```

Comment:
interesting

Reply:
This is required by the test I added - previously it was storing the entire object which made it ok to not store it in LKI, but now that I changed it to use a MOR I needed the LKI there.